src/test/ui/impl-trait/issue-99073-2.stderr

```diff
@@ -0,0 +1,15 @@
+error[E0308]: mismatched types
+  --> $DIR/issue-99073-2.rs:9:22
+   |
+LL | fn test<T: Display>(t: T, recurse: bool) -> impl Display {
+   |                                             ------------ the expected opaque type
+LL |     let f = || {
+LL |         let i: u32 = test::<i32>(-1, false);
```

Comment:
Span is kinda lacking here but it's all we have

Reply:
Yea, this is currently a problem for all RPITs